compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmInlineClassLowering.kt

```diff
@@ -89,6 +90,12 @@ private class JvmInlineClassLowering(private val context: JvmBackendContext) : F
         return declaration
     }
 
+    private fun copyDefaultArguments(replacement: IrReplacementFunction) {
+        for ((oldParameter, newParameter) in replacement.valueParameterMap) {
+            newParameter.defaultValue = oldParameter.owner.defaultValue?.deepCopyWithSymbols(replacement.function)
```

Comment:
Since `deepCopyWithSymbols` doesn't copy attributes, wouldn't it cause problems if an anonymous class is used in such a default argument, since we wouldn't have a name for it in the codegen?
E.g.
```
inline class A(val i: Int)

fun A.f(x: String = object { val v = "OK" }.v): String = x
```

Reply:
It does copy *attributes*, so there are no problems with names. You're probably thinking of metadata, and yeah, the object is missing some stuff compared to one in a similar `fun Any?.f`. Though note that `parameter.copyTo` calls `deepCopyWithSymbols` on the default value anyway, so that's not the fault of this PR specifically.

Reply:
Looks like the copy is redundant, though: `transformSimpleFunctionFlat` only returns two functions if the function is an override, but then it cannot have default arguments.

Reply:
Removed the copy and added a test.

Reply:
Oh you're right. But indeed, the copy is not necessary. Thanks!